crypto/asn1/a_mbstr.c

```diff
@@ -272,12 +270,15 @@ static int out_utf8(unsigned long value, void *arg)
 static int type_str(unsigned long value, void *arg)
 {
     unsigned long types;
+    const int native = ossl_fromascii(value);
```

Comment:
There is a *formal* ambiguity here in implicit conversion from "unsigned long" to "int". We already established that `value` is limited by 32 bits, *but* if most significant bit is set, then cast is rendered *formally* undefined. Do note "formally". From any practical angle it would work, but this side note is about *formal* compliance.